### PR TITLE
Use cloudgov-style sidenav for organizations and spaces

### PR DESCRIPTION
**Pull request by jeremiak**
To address #192, this uses the `cloudgov-style` assets to style the sidenav. Includes modifying the Navbar React component to use the correct HTML structure and data, updating the API utility module to also get all space information, and toggle the display of menus in the sidenav. All newly added code should have appropriate testing.

This is the first component that has some specific UI related state so we didn't have a pattern for handling that. After some discussion, @msecret and I decided to have UI actions that change state create actions to be dispatched and handled by the stores so that components can be as stateless as possible. To accommodate this, I added a new type of action creator that dispatches `UI_ACTION` sourced actions. This pattern was utilized to toggle the displays of submenus in the navigation. The component creates an `ORG_TOGGLE_SPACE_MENU` type action that the OrgStore registers to handle. The OrgStore adds a `space_menu_open` key to each organization and toggles it when told to.

Keeping this data in the store enables us to keep the components as stateless as possible. There was some talk about nesting all UI related state elements under a key like `ui` in each object, but we decided against it for the time being.

### CODE REVIEW

File: static_src/app.jsx [hunk 1]
```diff
@@ -23,6 +25,10 @@ export default class App extends React.Component {
     LoginStore.addChangeListener(this._onChange);
   }
 
+  componentWillUnmount() {
+    LoginStore.removeChangeListener(this._onChange);
```
Comment (msecret) [inline static_src/app.jsx]: 👍

File: static_src/app.jsx [hunk 2]
```diff
@@ -53,22 +56,29 @@ export default class App extends React.Component {
...
-      <div className="container-fluid">
-        <div className="row">
-          <nav className={ sidebarClasses }>
-            { sidebar }
-          </nav>
-          <main className={ mainClasses }>
-            { content }
-          </main>
+      <div className={ titleBarStyles['title_bar'] }>
+        <div class="nav_toggle">
```
Comment (msecret) [inline static_src/app.jsx]: Aren't these `class`es not actually being applied in React? Can they be removed?

File: static_src/app.jsx [hunk 3]
```diff
@@ -53,22 +56,29 @@ export default class App extends React.Component {
...
+          <i class="fa fa-bars nav_toggle-icon"></i>
+          <div class="icon-reorder tooltips" data-original-title="Toggle Navigation" data-placement="bottom">
```
Comment (msecret) [inline static_src/app.jsx]: Are those data attributes valid JSX code?

File: static_src/app.jsx (hunk 2 above)
Comment (jeremiak) [inline static_src/app.jsx]: I think we want them so I'll change the attribute to `className` as React expects

File: static_src/app.jsx (hunk 3 above)
Comment (jeremiak) [inline static_src/app.jsx]: Ya, apparently data attributes are really the only valid HTML and JSX attributes.

[Relevant JSX documentation](https://facebook.github.io/react/docs/jsx-gotchas.html#custom-html-attributes)

**Comment by msecret** [comment]
Lgtm after that one last class thing

**Comment by jeremiak** [comment]
@msecret You're right, I removed the FA classes

**Pull request merged**